include/wil/registry.h

```diff
@@ -17,16 +17,2704 @@
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.open_key(path, hkey.addressof(), access);
+        }
+
+        /**
+         * \brief Creates a new HKEY to the specified path - see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param[out] hkey A reference to a wil::shared_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT create_shared_key_nothrow(HKEY key, _In_opt_ PCWSTR path, _Out_::wil::shared_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            hkey.reset();
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(path, hkey.addressof(), access);
+        }
+#endif // #define __WIL_WINREG_STL
+
+#if defined(WIL_ENABLE_EXCEPTIONS)
+        /**
+         * \brief Queries for number of sub-keys
+         * \param key The HKEY to query for number of sub-keys
+         * \return The queried number of sub-keys if succeeded
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures
+         */
+        inline size_t get_child_key_count(HKEY key)
+        {
+            DWORD numSubKeys{};
+            THROW_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+                key, nullptr, nullptr, nullptr, &numSubKeys, nullptr,
+                nullptr, nullptr, nullptr, nullptr, nullptr, nullptr));
+            return numSubKeys;
+        }
+
+        /**
+         * \brief Queries for number of values
+         * \param key The HKEY to query for number of values
+         * \return The queried number of value if succeeded
+         * \exception std::exception (including wil::ResultException) will be thrown on all failures
+         */
+        inline size_t get_child_value_count(HKEY key)
+        {
+            DWORD numSubValues{};
+            THROW_IF_WIN32_ERROR(::RegQueryInfoKeyW(
+                key, nullptr, nullptr, nullptr, nullptr, nullptr,
+                nullptr, &numSubValues, nullptr, nullptr, nullptr, nullptr));
+            return numSubValues;
+        }
+#endif // #if defined(WIL_ENABLE_EXCEPTIONS)
+
+        /**
+         * \brief Queries for number of sub-keys
+         * \param key The HKEY to query for number of sub-keys
+         * \param[out] numSubKeys A pointer to a DWORD to receive the returned count
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT get_child_key_count_nothrow(HKEY key, _Out_ DWORD* numSubKeys) WI_NOEXCEPT

```

Comment:
No last modified? That's the one Info case we use I don't see covered

```
HRESULT get_last_modified_nothrow(HKEY key, _Out_ FILETIME* lastModified) WI_NOEXCEPT
{
    RETURN_IF_WIN32_ERROR(::RegQueryInfoKeyW(key,
        nullptr, nullptr, nullptr, nullptr, nullptr,
        nullptr, nullptr, nullptr, nullptr, nullptr, lastModified));
    return S_OK;
}
```

and the throwing wrapper of course

Reply:
Added. Thanks!

Reply:
`get_key_last_modified` or `get_last_modified`?

Reply:
Sure. that name is better.

Reply:
Note that I don't think we have tests for this.